engine/backends/proverif/proverif_backend.ml

```diff
@@ -8,29 +8,30 @@ include
       open Features
       include Off
       include On.Macro
+      include On.Question_mark
+      include On.Early_exit
     end)
     (struct
       let backend = Diagnostics.Backend.ProVerif
     end)
 
 module SubtypeToInputLanguage
     (FA : Features.T
-    (*   with *)
-    (*  type loop = Features.Off.loop *)
-    (* and type for_loop = Features.Off.for_loop *)
-    (* and type for_index_loop = Features.Off.for_index_loop *)
-    (* and type state_passing_loop = Features.Off.state_passing_loop *)
-    (* and type continue = Features.Off.continue *)
-    (* and type break = Features.Off.break *)
-    (* and type mutable_variable = Features.Off.mutable_variable *)
-    (* and type mutable_reference = Features.Off.mutable_reference *)
-    (* and type mutable_pointer = Features.Off.mutable_pointer *)
-    (* and type reference = Features.Off.reference *)
-    (* and type slice = Features.Off.slice *)
-    (* and type raw_pointer = Features.Off.raw_pointer *)
-    (* and type early_exit = Features.Off.early_exit *)
-    (* and type question_mark = Features.Off.question_mark *)
-    (* and type macro = Features.On.macro *)
+          (*  type loop = Features.Off.loop *)
+          (* and type for_loop = Features.Off.for_loop *)
+          (* and type for_index_loop = Features.Off.for_index_loop *)
+          (* and type state_passing_loop = Features.Off.state_passing_loop *)
+          (* and type continue = Features.Off.continue *)
+          (* and type break = Features.Off.break *)
+          (* and type mutable_variable = Features.Off.mutable_variable *)
+          (* and type mutable_reference = Features.Off.mutable_reference *)
+          (* and type mutable_pointer = Features.Off.mutable_pointer *)
+          (* and type reference = Features.Off.reference *)
+          (* and type slice = Features.Off.slice *)
+          (* and type raw_pointer = Features.Off.raw_pointer *)
+            with type early_exit = Features.On.early_exit
+             and type question_mark = Features.On.question_mark
+             and type macro = Features.On.macro
```

Comment:
I included these two because the example contains a `hacspec` macro and I think would not extract otherwise. Perhaps one of the two is not really necessary? In any case, when `hacspec_lib` is out of the example I probably won't need this.

Reply:
I've made a new issue to remind myself to check on the necessary features and restrictions on inputs, when translation of "known good" inputs works reliably. (#454)